packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	describeCompat,
+	TestDataObjectType,
+	type ITestDataObject,
+} from "@fluid-private/test-version-utils";
+import { type IContainerRuntimeOptions } from "@fluidframework/container-runtime/internal";
+import type { IContainerRuntimeBase } from "@fluidframework/runtime-definitions/internal";
+import { MockLogger } from "@fluidframework/telemetry-utils/internal";
+import {
+	type ITestObjectProvider,
+	createSummarizer,
+	createTestConfigProvider,
+	summarizeNow,
+} from "@fluidframework/test-utils/internal";
+
+import { TestSnapshotCache } from "../../testSnapshotCache.js";
+
+describeCompat("Odsp Network calls", "NoCompat", (getTestObjectProvider) => {
+	// Allow us to control summaries
+	const runtimeOptions: IContainerRuntimeOptions = {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				state: "disabled",
+			},
+		},
+	};
+	const configProvider = createTestConfigProvider({
+		"Fluid.Container.UseLoadingGroupIdForSnapshotFetch2": true,
+		"Fluid.Container.enableOfflineLoad": true,
+	});
+
+	let provider: ITestObjectProvider;
+	const testSnapshotCache = new TestSnapshotCache();
+
+	beforeEach("setup", async function () {
+		provider = getTestObjectProvider({ persistedCache: testSnapshotCache });
+		if (provider.driver.type === "odsp") {
+			this.skip();
+		}
+	});
+
+	const loadingGroupId = "loadingGroupId";
+	const createDataObjectsWithGroupIds = async (
+		mainObject: ITestDataObject,
+		containerRuntime: IContainerRuntimeBase,
+	) => {
+		const dataStoreA = await containerRuntime.createDataStore(
+			TestDataObjectType,
+			loadingGroupId,
+		);
+		const dataStoreB = await containerRuntime.createDataStore(
+			TestDataObjectType,
+			loadingGroupId,
+		);
+
+		mainObject._root.set("dataObjectA", dataStoreA.entryPoint);
+		mainObject._root.set("dataObjectB", dataStoreB.entryPoint);
+	};
+
+	it("Should not make odsp network calls", async () => {
```

Comment:
Also, please skip this as we know it is not passing. I will enable it with my PR.

Reply:
OK sounds good